### PR TITLE
CODENVY-334 Handle hosted workspace stop dialogs

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Show special dialog when workspace has been stopped due to inactivity

### CODE REVIEW

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml [hunk 1]
```diff
@@ -15,7 +15,8 @@
     from Codenvy S.A..
 
 -->
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (sleshchenko) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: mvn sortpom:sort?

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml [hunk 2]
```diff
@@ -25,8 +26,16 @@
...
     <packaging>jar</packaging>
     <name>Codenvy Plugin :: Hosted Environment Tuning</name>
+    <properties>
+        <maven.compiler.source>1.7</maven.compiler.source>
+        <maven.compiler.target>1.7</maven.compiler.target>
+    </properties>
     <dependencies>
         <dependency>
+            <groupId>com.codenvy.hosted-infrastructure</groupId>
+            <artifactId>codenvy-hosted-workspace-activity</artifactId>
```
Comment (sleshchenko) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: Why do you add server dependency to client module?

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml [hunk 3]
```diff
@@ -80,5 +97,22 @@
...
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>analyze</id>
+                        <configuration>
+                            <ignoredDependencies>
+                                <ignoredDependency>com.codenvy.hosted-infrastructure:codenvycodenvy-hosted-workspace-activity
```
Comment (sleshchenko) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: See https://github.com/codenvy/codenvy/pull/66/files#r58487123

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml (hunk 2 above)
Comment (mkuznyetsov) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: Server dependency contains constant of attribute

Comment (sleshchenko) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: It is wrong way to use server's constant from client code. You should think about creating shared module or place your constant in other common place

Comment (sleshchenko) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: It is wrong way to use server's constant from client code. You should think about creating shared module or place your constant in other common place

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/HostedWorkspaceStoppedHandler.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        };
+        eventBus.addHandler(WorkspaceStoppedEvent.TYPE, this);
+    }
+
+    @Override
+    public void onWorkspaceStopped(WorkspaceStoppedEvent event) {
+        workspaceServiceClient.getWorkspace(event.getWorkspace().getId()).then(new Operation<WorkspaceDto>() {
+            @Override
+            public void apply(WorkspaceDto workspace) throws OperationException {
+                if (workspace.getAttributes().get(WORKSPACE_STOPPED_DUE_TO_INACTIVITY).equals("true")) {
```
Comment (sleshchenko) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/HostedWorkspaceStoppedHandler.java]: Can this attribute be absent here?

Comment (sleshchenko) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/HostedWorkspaceStoppedHandler.java]: Can this attribute be absent here?

**Comment by skabashnyuk** [comment]
@vparfonov ?

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml (hunk 2 above)
Comment (vparfonov) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: +1

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/HostedWorkspaceStoppedHandler.java (hunk 4 above)
Comment (vparfonov) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/HostedWorkspaceStoppedHandler.java]: NPE?

File: core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/server/WorkspaceActivityManager.java [hunk 5]
```diff
@@ -65,6 +72,13 @@ public WorkspaceActivityManager(@Named("machine.ws_agent.inactive_stop_timeout_m
             public void onEvent(WorkspaceStatusEvent event) {
                 switch (event.getEventType()) {
                     case RUNNING:
+                        try {
+                            Workspace workspace = workspaceManager.getWorkspace(event.getWorkspaceId());
+                            workspace.getAttributes().remove(WORKSPACE_STOPPED);
+                            workspaceManager.updateWorkspace(event.getWorkspaceId(), workspace);
+
+                        } catch (Exception ex) {
```
Comment (evoevodin) [inline core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/server/WorkspaceActivityManager.java]: Write at least log message

File: core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/shared/Constants.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package com.codenvy.activity.shared;
+
+/**
+ * Workspace activity constants
+ *
+ * @author Mihail Kuznyetsov
+ */
+public class Constants {
```
Comment (evoevodin) [inline core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/shared/Constants.java]: Define constants class as `final`

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml (hunk 2 above)
Comment (mkuznyetsov) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: I think it is not worth it to create a separate shared module for 2 constans.
How about I just place Constants class in separate package?

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/HostedWorkspaceStoppedHandler.java (hunk 4 above)
Comment (mkuznyetsov) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/HostedWorkspaceStoppedHandler.java]: fixed

File: core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/shared/Constants.java (hunk 6 above)
Comment (mkuznyetsov) [inline core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/shared/Constants.java]: ok

File: core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/server/WorkspaceActivityManager.java (hunk 5 above)
Comment (mkuznyetsov) [inline core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/server/WorkspaceActivityManager.java]: fixed

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml (hunk 1 above)
Comment (mkuznyetsov) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: sort :)

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml (hunk 2 above)
Comment (vparfonov) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/pom.xml]: ok for now but try found solution to avoid dependencies like that

**Comment by vparfonov** [comment]
ok

File: core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/server/WorkspaceActivityManager.java [hunk 7]
```diff
@@ -14,10 +14,13 @@
  */
 package com.codenvy.activity.server;
 
+import sun.rmi.runtime.Log;
```
Comment (sleshchenko) [inline core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/server/WorkspaceActivityManager.java]: Check that you need this import

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/WorkspaceNotRunningView.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+ * @author Mihail Kuznyetsov
+ */
+public interface WorkspaceNotRunningView {
+
+
+    interface ActionDelegate {
+        /** Performs some actions when user clicks on create workspace button. */
+        void onOpenDashboardButtonClicked();
+
+        /** Performs some actions when user clicks on predefined recipes field. */
```
Comment (sleshchenko) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/WorkspaceNotRunningView.java]: fix `predefined recipes field`

**Comment by sleshchenko** [comment]
OK for me

File: plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/WorkspaceNotRunningView.java (hunk 8 above)
Comment (mkuznyetsov) [inline plugins/plugin-hosted/codenvy-hosted-ext-hosted/src/main/java/com/codenvy/ide/hosted/client/workspace/WorkspaceNotRunningView.java]: fixed

File: core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/server/WorkspaceActivityManager.java (hunk 7 above)
Comment (mkuznyetsov) [inline core/codenvy-hosted-workspace-activity/src/main/java/com/codenvy/activity/server/WorkspaceActivityManager.java]: fixed

**Comment by akorneta** [comment]
ok

**Pull request merged**